File: web/src/main/java/top/bettercode/summer/web/config/WebConfiguration.java
package top.bettercode.summer.web.config;

import org.springframework.boot.autoconfigure.AutoConfigureBefore;
import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import top.bettercode.summer.web.support.gb2260.GB2260Controller;
import top.bettercode.summer.web.support.packagescan.PackageScanClassResolver;

/**
 * @author Peter Wu
 * @since 0.1.14
 */
@AutoConfigureBefore({JacksonAutoConfiguration.class})
@EnableConfigurationProperties({SummerWebProperties.class, JacksonExtProperties.class})
@Configuration(proxyBeanMethods = false)
public class WebConfiguration {

    @Bean
    public PackageScanClassResolver packageScanClassResolver(ApplicationContext applicationContext) {
        return new PackageScanClassResolver(applicationContext.getClassLoader());
    }

    @Bean
    public GB2260Controller gb2260Controller() {
        return new GB2260Controller();
    }
}

File: web/src/main/kotlin/top/bettercode/summer/web/config/WebConfiguration.kt
package top.bettercode.summer.web.config

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.EnableAspectJAutoProxy
import top.bettercode.summer.tools.lang.util.TimeUtil
import top.bettercode.summer.tools.lang.util.WatchdogUtil
import top.bettercode.summer.web.properties.JacksonExtProperties
import top.bettercode.summer.web.properties.SummerWebProperties
import top.bettercode.summer.web.support.packagescan.PackageScanClassResolver

/**
 * @author Peter Wu
 */
@EnableConfigurationProperties(JacksonExtProperties::class, SummerWebProperties::class)
@EnableAspectJAutoProxy(proxyTargetClass = true)
@Configuration(proxyBeanMethods = false)
class WebConfiguration {
    private val log: Logger = LoggerFactory.getLogger(WebConfiguration::class.java)

    init {
        try {
            TimeUtil.checkTime()
            WatchdogUtil.schedule {
                TimeUtil.checkTime()
            }
        } catch (e: Exception) {
            log.error("check time error", e)
        }
    }

    @Bean
    fun packageScanClassResolver(applicationContext: ApplicationContext): PackageScanClassResolver {
        return PackageScanClassResolver(applicationContext.classLoader)
    }

}
